### DISCUSSION TITLE
[Feature] group conversations

### DISCUSSION THREAD

**Discussion by fuzzmonkey** [chatwoot/chatwoot#8190] [Feature Requests]
hey! We're looking at integrating a chat system into our app and this looks like it would be a really great fit! Thanks for open sourcing! 

**Is your feature request related to a problem? Please describe.**
We are a mortgage lender looking to offer mortgages to groups of people. Ideally we'd like to be able to have group conversations, where an operator or customer could add another user to their conversation. 

**Describe the solution you'd like**
I've not looked at the code in too much detail but wondering how feasible this would be?

**Describe alternatives you've considered**
We've looked at paid alternatives like Intercom, i'd be great to build on something we can customise and push features upstream though.

**Additional context**

**Comment by sojan-official** [comment]
@fuzzmonkey  Thanks a lot for exploring chatwoot.

Our system is envisioned in a way that customer initiates a conversation and multiple agents work on it. The ability for multiple parties to chat from the customer side isn't something we are actively exploring 

You may be able to get your use case by making additional code changes on top of chatwoot.  Please let us know if there is anything we can do to help you.

**Comment by aadrian** [comment]
"Group support" might be also helpful for cases where the customer initiates (or must initiate) the conversation, but really also needs the help/support of colleagues in order to be able to interact with the target party/site. Unfortunately in many business scenarios - this is the case :)  .

Without this feature, for these cases, the "conversation" is always moved over more classic media like email or phone, thus making it much longer than it would be over chat.

A "Share this conversation session" would solve such cases quickly.

**Comment by sojan-official** [comment]
@fuzzmonkey  @aadrian   The team talked about this and we could see this feature being helpful for many use cases.  We might not have the bandwidth to take this up on our immediate releases. But we can explore this further, for the subsequent sprints. We welcome any effort from the community on this as well.

Meanwhile, our product team would really love to understand your specific use cases. If you guys can jump on a call that will be super helpful. 

https://calendly.com/chatwoot/

**Comment by RonWalker22** [comment]
Hey guys, I'm thinking about building this feature for my own use case.

So, I'll have a share link for the agent or the customer to add more people to the conversation.

**Comment by pranavrajs** [comment]
@RonWalker22 This sounds good, let us know if you need any help from our part.

**Comment by RonWalker22** [comment]
I looked it over, I would have to change too much. The current Conversation model is so tightly coupled to a single customer. I'm not sure if I have not enough time.

**Comment by RonWalker22** [comment]
I think I may have found a way to add this feature without too much pain. 

I'm going to create a join table, `Guest`, that links a `Contact`(customer) with a `Conversation`. I'll add more attributes to `Guest` if i need to. 

```ruby
  Guest.create!(contact_id: guest.id, conversation_id: conversation.id)
```

The `Conversation` model along with all of it's attributes don't have to change. 

Also the workflow for a contact(customer) who is first to initiate the conversation, will not change. But we will add a guest for any additional contacts(customers) who join the conversation.

I plan on moving forward with this approach. If anyone has a better idea, let me know.

**Comment by sojan-official** [comment]
@RonWalker22  this sounds aligned with our thoughts.  Probably on the naming part, let's try something like  `ConversationParticipants` ( better ideas are welcome ) since `Guest` sounds a bit generic.  But this sounds like a good direction.  
At a later stage, we could migrate the contact attribute from conversation entirely to the new table. We could also add an additional attribute `primary` to the participant's table to denote the contact who triggered the conversation. But let's go at this in iterations.

**Comment by RonWalker22** [comment]
Maybe `Interlocutor`?
https://en.wikipedia.org/wiki/Interlocutor_(linguistics)

Although I never heard of the word before, it perfectly describes the model. But, if we need to provide the definition to others, perhaps, it's not the best word to use. 

I'll stick with ` ConversationParticipants` for now

**Comment by sojan-official** [comment]
`Interlocutor` :D 
the wiki link was really handy

**Comment by iwex** [comment]
Any news regarding this issue?

**Comment by pranavrajs** [comment]
Moving this to feature discussion as the feature is not planned in the immediate roadmap.
